### PR TITLE
WIP : Hypre Variants

### PR DESCRIPTION
**Pull request by PaulMullowney**
Adding some variants and some options to enable in hypre build. This doesn't seem to work.

### CODE REVIEW

**Review by PaulMullowney — COMMENTED** [review]
The options that I try to append in __configure_args don't seem to get added to the hypre autotools configuration. Not sure why.

File: repos/exawind/packages/hypre/package.py [hunk 1]
```diff
@@ -6,6 +6,9 @@ class Hypre(bHypre):
 
     phases = ['autoreconf', 'distclean', 'configure', 'clean', 'build', 'install']
 
+    variant('gpu-aware-mpi', default=True, description='Use gpu-aware mpi')
+    variant('umpire', default=True, description='Use Umpire')
```
Comment (psakievich) [inline repos/exawind/packages/hypre/package.py]: You need a `depends_on('umpire', when='+umpire')` to get it in the DAG.

**Comment by PaulMullowney** [comment]
Are you guys ok with this? Can we merge this in to main?

**Comment by psakievich** [comment]
Starting the CI. @PaulMullowney does it contain everything you need?  What about those changes to the mini-app that we were chatting about on slack?

**Comment by PaulMullowney** [comment]
Those are already pushed to mini-app master.

**Review by psakievich — APPROVED** [review]

**Pull request merged**